src/charm.py

```diff
@@ -42,6 +46,8 @@ def __init__(self, *args: typing.Any):
         self.smtp = SmtpRequires(self)
         self.s3 = S3Requirer(self, relation_name="s3")
         self.ingress = IngressPerAppRequirer(self, port=8080)
+        self.oauth: OAuthRequirer | None = None
```

Comment:
since we're using `_get_oauth` to initialize the oauth requirer is this still needed?

Reply:
`OAuthRequirer` cannot be initialized twice in the charm, so this variable acts as a cache here.

Reply:
I see, do you think `@functools.cached_property` or `@functools.lru_cache` be better suited as a cache?

Reply:
I agree that it's a good idea. However, when I tried, as `_get_oauth` may return `None` if the conditions aren't met, the value then becomes locked to `None` and remains unchanged even if the conditions later change.